acquisition.py

```diff
@@ -86,7 +91,87 @@ def get_reward_batch(self, inputs_af_base):  # inputs_af = list of ...
         inputs = pad_sequence(inputs_af, batch_first=True, padding_value=0.0)
 
         self.load_best_proxy()
-        self.proxy.model.eval()
+        self.proxy.model.train()
         with torch.no_grad():
             outputs = self.proxy.model(inputs, input_afs_lens, None)
         return outputs
+
+
+class AcquisitionFunctionUCB(AcquisitionFunctionBase):
+    def __init__(self, config, proxy):
+        super().__init__(config, proxy)
+
+    def load_best_proxy(self):
+        super().load_best_proxy()
+
+    def get_reward_batch(self, inputs_af_base):  # inputs_af = list of ...
+        super().get_reward_batch(inputs_af_base)
+
+        inputs_af = list(map(torch.tensor, inputs_af_base))
+        input_afs_lens = list(map(len, inputs_af_base))
+        inputs = pad_sequence(inputs_af, batch_first=True, padding_value=0.0)
+
+        self.load_best_proxy()
+        self.proxy.model.train()
+        with torch.no_grad():
+            outputs = (
+                torch.hstack([self.proxy.model(inputs, input_afs_lens, None)])
+                .cpu()
+                .detach()
+                .numpy()
+            )
+
+        mean = np.mean(outputs, axis=1)
+        std = np.std(outputs, axis=1)
+        score = mean + self.config.acquisition.ucb.kappa * std
```

Comment:
Do we need to denormalize the mean ?  To me, there is no need, but we have to keep in mind (for further stats) that this is not the UCB score with the "oracle model" values, it is normalized.

Reply:
I'm sorry not sure I understand your question: where is the mean being denormalised here?

Reply:
Like in the previous_code : actual_mean= stat_std* current_mean + stat_mean

Reply:
Yeah I thought we discuss we should not denormalise it because we're ultimately gonna apply rhe reward/boltzmann function.